CharlieBackend.Business/Services/AttachmentService.cs

```diff
@@ -131,17 +131,25 @@ public async Task<Result<DownloadAttachmentDto>> DownloadAttachmentAsync(long at
 
         public async Task<Result<AttachmentDto>> DeleteAttachmentAsync(long attachmentId)
         {
+
             var attachment = await _unitOfWork.AttachmentRepository.GetByIdAsync(attachmentId);
 
             if (attachment == null)
             {
                 return Result<AttachmentDto>.GetError(ErrorCode.ValidationError,
                      "Attachement with id: " + attachmentId + " is not found");
             }
+            var role = _currentUserService.Role;
 
-            await _blobService.DeleteAsync(attachment.ContainerName);
-
-            await _unitOfWork.AttachmentRepository.DeleteAsync(attachmentId);
+            if ( (attachment.CreatedByAccountId == _currentUserService.AccountId) || (role == UserRole.Admin))
+            {
+                await _blobService.DeleteAsync(attachment.ContainerName);
+                await _unitOfWork.AttachmentRepository.DeleteAsync(attachmentId);
+            }
+            else
+            {
+                return Result<AttachmentDto>.GetError(ErrorCode.NotFound, "You cannot delete another user's data");
+            }
 
             await _unitOfWork.CommitAsync();
```

Comment:
I think we should commit changes only after deleting Attachment from the database, this part of the code should be inside the "if" statement

Reply:
ok